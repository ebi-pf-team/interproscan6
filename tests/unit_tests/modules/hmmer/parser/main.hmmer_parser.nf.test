nextflow_process {

    name "Test Process HMMER_PARSER"
    script "interproscan/modules/hmmer/parser/main.nf"
    process "HMMER_PARSER"

    stage {
        symlink "interproscan/scripts/hmmer/parser_out.py"
        symlink "tests/unit_tests/test_inputs/hmmer3_parser/gene3d_out"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer3_parser/gene3d_out")
                input[1] = "gene3d"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
