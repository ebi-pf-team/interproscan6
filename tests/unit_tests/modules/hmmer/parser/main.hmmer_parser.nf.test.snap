{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmer_parsed_gene3d_out.json:md5,d861a26e649cc145b9c828db01bc68dd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T08:47:23.419797"
    }
}